core/src/main/java/org/geysermc/geyser/registry/populator/CustomItemRegistryPopulator.java

```diff
@@ -390,13 +398,74 @@ private static void computeBlockItemProperties(String blockItem, NbtMapBuilder c
+    private static void computeChargeableProperties(NbtMapBuilder itemProperties, NbtMapBuilder componentBuilder, String mapping, int protocolVersion) {
         // setting high use_duration prevents the consume animation from playing
         itemProperties.putInt("use_duration", Integer.MAX_VALUE);
         // display item as tool (mainly for crossbow and bow)
         itemProperties.putBoolean("hand_equipped", true);
-        // ensure client moves at slow speed while charging (note: this was calculated by hand as the movement modifer value does not seem to scale linearly)
-        componentBuilder.putCompound("minecraft:chargeable", NbtMap.builder().putFloat("movement_modifier", 0.35F).build());
+        // Make bows, tridents, and crossbows enchantable
+        itemProperties.putInt("enchantable_value", 1);
+
+        if (GameProtocol.is1_20_60orHigher(protocolVersion)) {
+            componentBuilder.putCompound("minecraft:use_modifiers", NbtMap.builder()
+                    .putFloat("use_duration", 100F)
+                    .putFloat("movement_modifier", 0.35F)
+                    .build());
+
+            switch (mapping) {
+                case "minecraft:bow" -> {
+                    itemProperties.putString("enchantable_slot", "bow");
+                    itemProperties.putInt("frame_count", 3);
+
+                    componentBuilder.putCompound("minecraft:shooter", NbtMap.builder()
+                            .putList("ammunition", NbtType.COMPOUND, List.of(
+                                    NbtMap.builder()
+                                            .putCompound("item", NbtMap.builder()
+                                                    .putString("name", "minecraft:arrow")
+                                                    .build())
+                                            .putBoolean("use_offhand", true)
+                                            .putBoolean("search_inventory", true)
+                                            .build()
+                            ))
+                            .putFloat("max_draw_duration", 0f)
+                            .putBoolean("charge_on_draw", true)
+                            .putBoolean("scale_power_by_draw_duration", true)
+                            .build());
+                    componentBuilder.putInt("minecraft:use_duration", 999);
+                }
+                case "minecraft:trident" -> {
+                    itemProperties.putString("enchantable_slot", "trident");
+                    componentBuilder.putInt("minecraft:use_duration", 999);
+                }
+                case "minecraft:crossbow" -> {
+                    itemProperties.putString("enchantable_slot", "crossbow");
+                    itemProperties.putInt("frame_count", 10);
+                    //itemProperties.putInt("minecraft:use_animation", 4); // Todo: find proper cross bow animation value
```

Comment:
Wouldn't worry about this. Bows and crossbows with geyser will need attachables to function properly so no point adding extra hardcoded animation.

Reply:
ill yeet these then & merge